sound/soc/sdca/sdca_functions.c

```diff
@@ -119,6 +119,140 @@ static int find_sdca_entities(struct device *dev,
 	return 0;
 }
 
+static struct sdca_entity *find_sdca_entity_by_label(struct sdca_function *function,
+						     const char *label)
+{
+	int i;
+
+	for (i = 0; i < function->num_entities; i++) {
+		struct sdca_entity *entity;
+
+		entity = &function->entities[i];
+
+		if (!strcmp(entity->label, label))
+			return entity;
+	}
+
+	return NULL;
+}
+
+static int find_sdca_entity_connection(struct device *dev,
+				       struct fwnode_handle *function_node,
+				       struct sdca_function *function,
+				       struct fwnode_handle *entity_node,
+				       struct sdca_entity *entity)
+{
+	u64 input_pin_list;
+	int pin;
+
+	fwnode_property_read_u64(entity_node, "mipi-sdca-input-pin-list",
+				 &input_pin_list);
+
+	if (!input_pin_list)
+		return 0;
+
+	/*
+	 * Each bit set in the input-pin-list refers to an entity_id
+	 *  in this Function. Entity0 is an illegal connection since
+	 *  it is used for Function-level configurations
+	 */
+	if (input_pin_list & BIT(0)) {
+		dev_err(dev, "%s: %pfwP: entity_id %#x has invalid input_pin 0\n",
+			__func__, function_node, entity->id);
+		return -EINVAL;
+	}
+
+	for(pin = 1; pin < 64; pin++) {
+		struct fwnode_handle *connected_node;
+		struct sdca_entity *connected_entity;
+		const char *connected_label;
+		char pin_property[40];
```

Comment:
I would be tempted to add some sort of define for this max length, lot of max 40 byte strings used in the code.